libs/MeshKernel/include/MeshKernel/MeshInterpolationInterface.hpp

```diff
@@ -0,0 +1,76 @@
+//---- GPL ---------------------------------------------------------------------
+//
+// Copyright (C)  Stichting Deltares, 2011-2021.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+// contact: delft3d.support@deltares.nl
+// Stichting Deltares
+// P.O. Box 177
+// 2600 MH Delft, The Netherlands
+//
+// All indications and logos of, and references to, "Delft3D" and "Deltares"
+// are registered trademarks of Stichting Deltares, and remain the property of
+// Stichting Deltares. All rights reserved.
+//
+//------------------------------------------------------------------------------
+
+#pragma once
+
+namespace meshkernel
+{
+    /// @brief The interpolant types
+    enum class Interpolants
+    {
+        AveragingInterpolation = 0,
+        BilinearInterpolationOnGriddedSamples = 1,
+    };
+
+    /// @brief Interface for interpolation methods
+    class MeshInterpolationInterface
+    {
+    public:
+        /// @brief Virtual destructor
+        virtual ~MeshInterpolationInterface() = default;
+
+        /// @brief Compute
+        virtual void Compute() = 0;
```

Comment:
Only compute needs to be virtual. All other virtual functions can be implemented here, once the vectors `m_nodeResults`, `m_edgeResults` and `m_faceResults` are moved to the private scope of MeshInterpolationInterface.